src/views/open-api-docs-view/utils/get-property-detail-props.ts

```diff
@@ -3,31 +3,120 @@
  * SPDX-License-Identifier: MPL-2.0
  */
 
+// Utilities
+import markdownToHtml from '@hashicorp/platform-markdown-utils/markdown-to-html'
 // Types
 import type { OpenAPIV3 } from 'openapi-types'
-import type { PropertyDetailProps } from '../types'
+import type { PropertyDetailsProps } from '../components/property-details'
 
 /**
  * Given a parameter object,
  * Return property detail props.
- *
- * TODO: this is a placeholder, still needs to be implemented.
  */
-export function getPropertyDetailPropsFromParameter(
+export async function getPropertyDetailPropsFromParameter(
 	param: OpenAPIV3.ParameterObject
-): PropertyDetailProps {
-	return { name: param.name, _placeholder: param }
+): Promise<PropertyDetailsProps> {
+	const paramSchemaDetails = await getPropertyDetailsFromSchema(param.schema)
+	return {
+		name: param.name,
+		type: paramSchemaDetails.typeString,
+		// Note that we use the description from the parameter itself,
+		// rather than from the parameter's schema.
+		description: await markdownToHtml(param.description),
+		isRequired: param.required,
+		nestedProperties: paramSchemaDetails.nestedProperties,
+	}
 }
 
 /**
  * Given a schema object,
  * Return property detail props.
- *
- * TODO: this is a placeholder, still needs to be implemented.
  */
-export function getPropertyDetailPropsFromSchemaObject(
+export async function getPropertyDetailPropsFromSchemaObject(
+	key: string,
 	schema: OpenAPIV3.SchemaObject,
-	key: string
-): PropertyDetailProps {
-	return { name: key, _placeholder: schema }
+	isRequired?: boolean
+): Promise<PropertyDetailsProps> {
+	const schemaDetails = await getPropertyDetailsFromSchema(schema)
+	return {
+		name: key,
+		type: schemaDetails.typeString,
+		description: schemaDetails.description,
+		isRequired,
+		nestedProperties: schemaDetails.nestedProperties,
+	}
+}
+
+/**
+ * Given a schema object,
+ * Return a typeString summarizing the schema,
+ * descriptionHtml generated from the schema.description,
+ * and an array of nested properties with fully built props.
+ */
+async function getPropertyDetailsFromSchema(
```

Comment:
This is where the more interesting parsing happens, transforming parameters and schemas into the props we show on the front-end.